tests/test_pipeline_media.py

```diff
@@ -35,8 +31,26 @@ def _mocked_download_func(request, info):
     return response() if callable(response) else response
 
 
+class UserDefinedPipeline(MediaPipeline):
```

Comment:
I wouldn't reuse the old MediaPipeline implementations here but followed the expected return values. So:

* media_downloaded() should return {} (technically, that's also wrong but better than returning `response`)
* file_path() should return ''

Others I think are fine.

Reply:
done